### PR TITLE
Feature/opene 213

### PR DESCRIPTION
**Pull request by jrudinskij**

### CODE REVIEW

**Comment by jrudinskij** [comment]
Hi Arnas,

Could you please review when you get time?
Thanks.

File: openesdh-repo/src/main/java/dk/openesdh/repo/services/cases/CaseOwnersServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+                .collect(Collectors.toSet());
+    }
+    
+    private Set<String> getUserIds(NodeRef authorityNodeRef) {
+        QName type = nodeService.getType(authorityNodeRef);
+        if (type.isMatch(ContentModel.TYPE_AUTHORITY_CONTAINER)) {
+            String groupName = (String) nodeService.getProperty(authorityNodeRef, ContentModel.PROP_NAME);
+            return authorityService.getContainedAuthorities(AuthorityType.USER, groupName, false);
+        } else if (type.isMatch(ContentModel.TYPE_PERSON)) {
+            return new HashSet<String>(Arrays.asList(personService.getPerson(authorityNodeRef).getUserName()));
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/services/cases/CaseOwnersServiceImpl.java]: there is an util just for this :)
Sets.newHashSet(element1,...)

File: openesdh-repo/src/main/java/dk/openesdh/repo/services/members/CaseMembersServiceImpl.java [hunk 2]
```diff
@@ -98,6 +102,17 @@ public void addAuthorityToRole(final NodeRef authorityNodeRef, final String role
 
     @Override
     public void addAuthoritiesToRole(final List<NodeRef> authorities, final String role, final NodeRef caseNodeRef) {
+        List<String> authoritiesNames = authorities.stream()
+                .map(nodeRef -> Optional.ofNullable(getAuthorityName(nodeRef)))
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/services/members/CaseMembersServiceImpl.java]: you could loose Optional and two .map'ings if you use utill "Objects"

```
List<String> filterStrings2 = strings
            .stream()
            .filter(Objects::nonNull)
            .collect(Collectors.toList());
```

File: openesdh-repo/src/main/java/dk/openesdh/repo/services/xsearch/CaseDocumentsAttachmentsSearchServiceImpl.java [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        CaseDocumentsAttachmentsSearchService {
+
+    @Autowired
+    @Qualifier("DocumentService")
+    protected DocumentService documentService;
+
+    @Override
+    protected JSONObject nodeToJSON(NodeRef nodeRef) throws JSONException {
+        JSONObject node = super.nodeToJSON(nodeRef);
+        String fileName = (String) nodeService.getProperty(nodeRef, ContentModel.PROP_NAME);
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/services/xsearch/CaseDocumentsAttachmentsSearchServiceImpl.java:27]: isn't here PROP_TITLE should be used for documents?

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/authorities/UserGroupsWebScript.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+
+    private static final String SITE_GROUP_PREFIX = "GROUP_site_";
+    
+    private static final String SITE_GROUP_PREFIX2 = "GROUP_SITE_";
+
+    private static final String ALFRESCO_GROUP_PREFIX = "GROUP_ALFRESCO_";
+
+    private static final String EMAIL_GROUP_PREFIX = "GROUP_EMAIL_";
+
+    private static final List<String> nonCaseOwnersGroupPrefixes = new ArrayList<String>();
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/authorities/UserGroupsWebScript.java]: You could assign values directly

```
private static final List<String> nonCaseOwnersGroupPrefixes = Arrays.asList(....
```

File: openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/authorities/UserGroupsWebScript.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        String name = AuthenticationUtil.getFullyAuthenticatedUser();
+        Set<String> groups = authorityService.getContainingAuthorities(AuthorityType.GROUP, name, false)
+                .stream()
+                .filter(this::isCaseOwnersGroup)
+                .collect(Collectors.toSet());
+        return WebScriptUtils.jsonResolution(groups);
+    }
+
+    private boolean isCaseOwnersGroup(String groupName) {
+        return !nonCaseOwnersGroupPrefixes.stream()
```
Comment (ArnasPetraitis) [inline openesdh-repo/src/main/java/dk/openesdh/repo/webscripts/authorities/UserGroupsWebScript.java:54]: I would suggest to use regex here :)

**Comment by jrudinskij** [comment]
retest this please

**Pull request merged**